katrain/core/ai.py

```diff
@@ -60,10 +61,13 @@ def ai_move(game: Game, ai_mode: str, ai_settings: Dict) -> Tuple[Move, GameNode
         policy_grid = var_to_grid(cn.policy, size)  # type: List[List[float]]
         top_policy_move = policy_moves[0][1]
         ai_thoughts += f"Using policy based strategy, base top 5 moves are {fmt_moves(policy_moves[:5])}. "
+        len_legal_policy_moves = len([(pol, mv) for pol, mv in policy_moves if not mv.is_pass if pol > 0])
         if ai_mode == AI_POLICY and cn.depth <= ai_settings["opening_moves"]:
             ai_mode = AI_WEIGHTED
             ai_thoughts += f"Switching to weighted strategy in the opening {int(ai_settings['opening_moves'])} moves. "
             ai_settings = {"pick_override": 0.9, "weaken_fac": 1, "lower_bound": 0.02}
+        if ai_mode == AI_RANK:
+            ai_settings = {"pick_override": (0.8*(1-(361-len_legal_policy_moves)/361.*.5)), "kyu": ai_settings["kyu"] }
```

Comment:
You realize katrain can play anything from 1x2 to 52x52 right? so hardcoding 361 is not prefered
also the . in 361. is redundant.